modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItFloatingPointTest.java

```diff
@@ -0,0 +1,190 @@
+
+        assertEquals(Double.NEGATIVE_INFINITY, sql("SELECT '-Infinity'::DOUBLE - 1").get(0).get(0));
+        assertEquals(Double.NEGATIVE_INFINITY, sql("SELECT '-Infinity'::DOUBLE + 1").get(0).get(0));
+
+        assertEquals(Double.POSITIVE_INFINITY, sql("SELECT '+Infinity'::DOUBLE - 1").get(0).get(0));
+        assertEquals(Double.POSITIVE_INFINITY, sql("SELECT '+Infinity'::DOUBLE + 1").get(0).get(0));
+
+        assertEquals(Double.NaN, sql("SELECT 'NaN'::DOUBLE - 1").get(0).get(0));
+        assertEquals(Double.NaN, sql("SELECT '-NaN'::DOUBLE + 1").get(0).get(0));
+        assertEquals(Double.NaN, sql("SELECT '-Infinity'::DOUBLE + '+Infinity'::DOUBLE").get(0).get(0));
+        assertEquals(Double.NaN, sql("SELECT '-Infinity'::DOUBLE / '+Infinity'::DOUBLE").get(0).get(0));
+    }
+
+    @Test
+    void testLiterals() {
+        assertEquals(-0.0f, sql("SELECT -0.0::FLOAT").get(0).get(0));
+        assertEquals(0.0f, sql("SELECT 0.0::FLOAT").get(0).get(0));
+        assertEquals(Float.NEGATIVE_INFINITY, sql("SELECT '-Infinity'::FLOAT").get(0).get(0));
+        assertEquals(Float.POSITIVE_INFINITY, sql("SELECT '+Infinity'::FLOAT").get(0).get(0));
+        assertEquals(Float.NaN, sql("SELECT 'NaN'::FLOAT").get(0).get(0));
+        assertEquals(Float.NaN, sql("SELECT -'NaN'::FLOAT").get(0).get(0));
+
+        assertEquals(-0.0d, sql("SELECT -0.0::DOUBLE").get(0).get(0));
+        assertEquals(0.0d, sql("SELECT 0.0::DOUBLE").get(0).get(0));
+        assertEquals(Double.NEGATIVE_INFINITY, sql("SELECT '-Infinity'::DOUBLE").get(0).get(0));
+        assertEquals(Double.POSITIVE_INFINITY, sql("SELECT '+Infinity'::DOUBLE").get(0).get(0));
+        assertEquals(Double.NaN, sql("SELECT 'NaN'::DOUBLE").get(0).get(0));
+        assertEquals(Double.NaN, sql("SELECT -'NaN'::DOUBLE").get(0).get(0));
+    }
+
+    @ParameterizedTest
+    @ValueSource(floats = {
+            -0.0f,
+            0.0f,
+            Float.NEGATIVE_INFINITY,
+            Float.POSITIVE_INFINITY,
+            Float.NaN
+    })
+    void testParameters(float f) {
+        assertEquals(f, (float) sql("SELECT ?", f).get(0).get(0));
+    }
+
+    @ParameterizedTest
+    @ValueSource(doubles = {
+            -0.0d,
+            0.0d,
+            Double.NEGATIVE_INFINITY,
+            Double.POSITIVE_INFINITY,
+            Double.NaN
+    })
+    void testParameters(double f) {
+        assertEquals(f, (double) sql("SELECT ?", f).get(0).get(0));
+    }
+
+    @Test
+    void testOrderBy() {
+        List<Float> floats = sql("SELECT f FROM test ORDER BY f")
+                .stream().flatMap(List::stream).map(Float.class::cast).collect(toList());
+        List<Double> doubles = sql("SELECT d FROM test ORDER BY d")
+                .stream().flatMap(List::stream).map(Double.class::cast).collect(toList());
+
+        assertThat(floats, contains(Float.NEGATIVE_INFINITY, -1.0f, -0.0f, 0.0f, 1.0f, Float.POSITIVE_INFINITY, Float.NaN));
```

Comment:
force_index ? if we want to check IndexScan too

Reply:
Fixed